test/units/module_utils/facts/test_collectors.py

```diff
@@ -238,6 +238,46 @@ def test_collect(self):
         self.assertIn('pkg_mgr', facts_dict)
 
 
+class TestMacOSXPkgMgrFacts(BaseFactsTest):
+    __test__ = True
+    gather_subset = ['!all', 'pkg_mgr']
+    valid_subsets = ['pkg_mgr']
+    fact_namespace = 'ansible_pkgmgr'
+    collector_class = PkgMgrFactCollector
+    collected_facts = {
+        "ansible_distribution": "MacOSX",
+        "ansible_distribution_major_version": "11",
+        "ansible_os_family": "Darwin"
+    }
+
+    @patch('ansible.module_utils.facts.system.pkg_mgr.os.path.exists', side_effect=lambda x: x == '/opt/homebrew/bin/brew')
```

Comment:
I don't think the `lambda` is necessary here. You should be able to set [`return_value`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.return_value) instead.

Reply:
@samdoran Thanks for reviewing. How would you set `return_value` to be `True` only for that specific path and not for the others?

Reply:
Oh, I misread this. I thought it was returning the path not the result of an evaluation. I tend to try and avoid `lambda` at all costs (harder to read/maintain usually), but I don't see a better way to write the test.

Reply:
Makes sense, thanks for checking. I thought this was the most concise way to do it.